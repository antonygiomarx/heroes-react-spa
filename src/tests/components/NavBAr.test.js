const { mount } = require("enzyme");
const { MemoryRouter, Router } = require("react-router-dom");
const { AuthContext } = require("../../auth/authContext");
const { NavBar } = require("../../components/NavBar/NavBar");
const { types } = require("../../types/types");

describe('Test in the NavBa Component', () => {
    
    const historyMock = {
        replace: jest.fn(),
        push: jest.fn(),
        location: {},
        listen: jest.fn(),
        createHref: jest.fn()
    }

    const contextValue = {
        dispatch: jest.fn(),
        user: {
            logged: true,
            name: 'Pedrinho'
        }
    }
   
    const wrapper = mount(
        <AuthContext.Provider value={ contextValue }>
            <MemoryRouter>
                <Router history={ historyMock }>
                    <NavBar/>
                </Router>
            </MemoryRouter>
        </AuthContext.Provider>
    )
    
    const divUser = wrapper.find('.user')

    afterEach( () => {
        jest.clearAllMocks();
    })
    


    test('should show itself correctly', () => {
        
        
        expect( wrapper ).toMatchSnapshot();
        expect( divUser.find('span').text().trim() ).toBe( 'Pedrinho' )
    });

    test('should call the logout and use history prop', () => {
        
        divUser.find('button').simulate('click')
         
        expect( historyMock.replace ).toHaveBeenCalledWith("/login")
        expect( contextValue.dispatch ).toHaveBeenCalledWith( { type: types.logout } )
        
    });


});