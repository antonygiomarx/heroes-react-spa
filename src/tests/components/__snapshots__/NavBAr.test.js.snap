// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in the NavBa Component should show itself correctly 1`] = `
<nav>
  <div
    className="navbar__header"
  >
    <header>
      <a
        className="icon__link"
        onClick={[Function]}
      >
         
        <h1>
           HeroesApp 
        </h1>
         
      </a>
    </header>
  </div>
  <div
    className="nav-menu-active "
  >
    <ul
      className="navbar"
    >
      <li
        className="nav__item"
      >
        <a
          aria-current={null}
          className="nav__link"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="nav__item"
      >
        <a
          aria-current={null}
          className="nav__link"
          onClick={[Function]}
        >
          About
        </a>
      </li>
      <li
        className="nav__item"
      >
        <a
          aria-current={null}
          className="nav__link"
          onClick={[Function]}
        >
          Marvel
        </a>
      </li>
      <li
        className="nav__item"
      >
        <a
          aria-current={null}
          className="nav__link"
          onClick={[Function]}
        >
          DC
        </a>
      </li>
      <li
        className="nav__item"
      >
        <a
          aria-current={null}
          className="nav__link"
          onClick={[Function]}
        >
          Search
        </a>
      </li>
    </ul>
  </div>
  <div
    className="btn__toggle"
  >
    <button
      onClick={[Function]}
    >
      <svg
        className="react-icons"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 352 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
          key="0"
        />
      </svg>
    </button>
  </div>
  <div
    className="user"
  >
    <span>
      Pedrinho
    </span>
    <button
      className="logout__btn"
      onClick={[Function]}
    >
      Log out
    </button>
  </div>
</nav>
`;
